doc/specs/#140 - ZIP Support.md

```diff
@@ -0,0 +1,90 @@
+
+## Design Flow/Implementation
+
+Because ZIP archive files are essentially a compressed directory containing the relevant installer files, the install flow will need to add the following steps:
+
+1. Perform checks on the downloaded ZIP archive for potential malware or threats.
+2. Extract the contents of the ZIP archive to a temporary location.
+3. Execute the appropriate install flow using the extracted installer based on the installer type derived from the manifest.
+
+>NOTE: The initial implementation will first support installing ZIPs that contain only basic installers (msix, msi, or exe), then portable apps.
+
+## Manifest Changes:
+Addition of `NestedInstallerType`:
+- Enumeration of supported nested installerTypes contained inside an archive file.
+
+Addition of `NestedInstallerFile`
+- Object containing metadata about a nested installer file contained inside an archive. 
+- Properties:
+    - `RelativeFilePath`: The relative path to a nested installer file contained inside an archive.
+    - `PortableCommandAlias`: The command alias to be used for calling the package. Only applies to a nested portable package.
+
+These changes would be added to the `Installer` object with `NestedInstallerFiles` representing an array of `NestedInstallerFile`. 
+
+### Manifest Validation:
+- Exactly one `NestedInstallerFile` entry must be specified for a non-portable nested installer type.
+- Multiple `NestedInstallerFile` entries can be specified for portable nested installer type to support a suite of portable exes that would all be installed under the same package.
+- If a ZIP `InstallerType` is specified, a `NestedInstallerType` is required to be specified.
+
+## Supported Install Scenarios:
+The initial implementation of this feature will only support the following installation scenarios:
+- Single base installer (exe, msi, msix) contained inside an archive file. 
+- Single or multiple portable exes bundled as a suite inside an archive file.
+
+## ZIP Extraction
+The extraction of ZIPs will be done using Windows Shell APIs. ZIP files can be represented as a [ShellFolder](https://docs.microsoft.com/en-us/windows/win32/api/shobjidl_core/nn-shobjidl_core-ishellfolder) object that can be used to manage the contents of the ZIP file. File contents are represented as [ShellItems](https://docs.microsoft.com/en-us/windows/win32/api/shobjidl_core/nn-shobjidl_core-ishellitem), which can be handled using the methods exposed by the [IFileOperation interface](https://docs.microsoft.com/en-us/windows/win32/api/shobjidl_core/nn-shobjidl_core-ifileoperation). If the `NestedInstallerType` is non-portable, the installer file will be copied to a temporary location and used during execution of the installation flow. If the `NestedInstallerType` is portable, all files will be moved to the appropriate install location and symlinks will be created for each `NestedInstallerFile` entry.

```

Comment:
> If the `NestedInstallerType` is non-portable, the installer file will be copied to a temporary location and used during execution of the installation flow.

Are we going to support setup.exe with data cabs beside it? I thought this will be the most used cases for zip. In this case, only copying the installer file is not enough.

Reply:
Yes, it should be "the entire archive is extracted to a temporary location".

Reply:
Changed to specify that the entire archive will be extracted to a temporary location